test/test_torch.py

```diff
@@ -4147,6 +4153,86 @@ def test_serialization_offset(self):
         self.assertEqual(i, i_loaded)
         self.assertEqual(j, j_loaded)
 
+    @unittest.skipIf(
+        not TEST_DILL or not check_module_version_greater_or_equal(dill, (0, 3, 1)),
+        '"dill" not found or not correct version'
+    )
+    def test_serialization_dill_version_supported(self):
+        x = torch.randn(5, 5)
+
+        with tempfile.NamedTemporaryFile() as f:
+            try:
+                torch.save(x, f, pickle_module=dill)
+            except ValueError:
+                error_was_raised = True
+            else:
+                error_was_raised = False
+
+            self.assertFalse(error_was_raised)
+
+            f.seek(0)
+            try:
+                x2 = torch.load(f, pickle_module=dill, encoding='utf-8')
+            except ValueError:
+                error_was_raised = True
+            else:
+                error_was_raised = False
+            self.assertFalse(error_was_raised)
+
+    @unittest.skipIf(
+        not TEST_DILL or check_module_version_greater_or_equal(dill, (0, 3, 1)),
+        '"dill" not found or is correct version'
+    )
+    def test_serialization_dill_version_not_supported(self):
+        x = torch.randn(5, 5)
+
+        with tempfile.NamedTemporaryFile() as f:
+            try:
+                torch.save(x, f, pickle_module=dill)
+            except ValueError:
+                error_was_raised = True
+            else:
+                error_was_raised = False
+
+            self.assertTrue(error_was_raised)
+
+            f.seek(0)
+            try:
+                x2 = torch.load(f, pickle_module=dill, encoding='utf-8')
+            except ValueError:
+                error_was_raised = True
+            else:
+                error_was_raised = False
+            self.assertTrue(error_was_raised)
+
+    @unittest.skipIf(
+        not TEST_DILL or not check_module_version_greater_or_equal(dill, (0, 3, 1)),
+        '"dill" not found or not correct version'
+    )
+    def test_serialization_dill_no_encoding(self):
+        x = torch.randn(5, 5)
+
+        with tempfile.NamedTemporaryFile() as f:
+            torch.save(x, f, pickle_module=dill)
+            f.seek(0)
+            x2 = torch.load(f, pickle_module=dill)
+            self.assertIsInstance(x2, type(x))
+            self.assertEqual(x, x2)
+
+    @unittest.skipIf(
+        not TEST_DILL or not check_module_version_greater_or_equal(dill, (0, 3, 1)),
+        '"dill" not found or not correct version'
+    )
+    def test_serialization_dill_encoding(self):
```

Comment:
nit: I feel like we can merge two of test_serialization_dill_supported, test_serialization_dill_encoding, and test_serialization_dill_no_encoding. But it's nice to have them as separate test cases as well